src/backend/base/langflow/api/v1/folders.py

```diff
@@ -89,25 +92,41 @@ def read_folders(
                 or_(Folder.user_id == current_user.id, Folder.user_id == None)  # noqa: E711
             )
         ).all()
+        folders = [folder for folder in folders if folder.name != STARTER_FOLDER_NAME]
         return sorted(folders, key=lambda x: x.name != DEFAULT_FOLDER_NAME)
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e)) from e
 
 
-@router.get("/{folder_id}", response_model=FolderReadWithFlows, status_code=200)
+@router.get("/{folder_id}", response_model=FolderWithPaginatedFlows, status_code=200)
 def read_folder(
     *,
     session: Session = Depends(get_session),
     folder_id: str,
     current_user: User = Depends(get_current_active_user),
+    params: Params = Depends(),
+    is_component: bool = False,
+    is_flow: bool = False,
+    search: str = "",
 ):
     try:
         folder = session.exec(select(Folder).where(Folder.id == folder_id, Folder.user_id == current_user.id)).first()
         if not folder:
             raise HTTPException(status_code=404, detail="Folder not found")
-        flows_from_current_user_in_folder = [flow for flow in folder.flows if flow.user_id == current_user.id]
-        folder.flows = flows_from_current_user_in_folder
-        return folder
+
+        stmt = select(Flow).where(Flow.folder_id == folder_id, Flow.user_id == current_user.id)
```

Comment:
Why do we make a separate db call, if the `folder` already contains the `flows`?

Reply:
Also.- if the first db statement already filters to a specific user id, we don't need to do that again, right? Or can there be a distinction between a Folder Owner and a Flow Owner?

Reply:
actually I'm not joining the flow inside the folder call first.
I just need to check if the folder exists.